HW-12/main.py

```diff
@@ -0,0 +1,163 @@
+# 1. Використати файл як базу даних для збереження записів телефонної книги із попередніх завдань.
+#
+# Ваша телефонна книга, що до цього містилася в dict, має зберігатися у вигляді тексту в JSON форматі.
+# При закритті програми і повторному відкритті всі попередні дані мають бути доступними.
+# Підказка: Ви можете використати бібліотеку json, яка допоможе із перетворенням dict в JSON string (при записі в файл) і JSON string в dict (при читанні із файлу). Файл слід оновлювати після кожної успішної операції add або delete.
+
+import json
+import time
+
+
+def number_validation(number):
+    split_plus = number.split("+")
+    if not number.startswith("+"):
+        print("Number must starts with '+'")
+        return False
+    elif not split_plus[1].isdigit():
+        print("Must be integer")
+        return False
+    elif len(split_plus[1]) != 12:
+        print('Length is not correct')
+        return False
+    else:
+        return True
+
+
+phonebook = {}
+
+with open('phonebook.json', 'r') as file:
+    phonebook = json.load(file)
+
+print(f'Now in your contacts: {phonebook}')
+
+command = ''
+
+while True:
+
+    if command == '':
+        command = input('Enter command here: ')
+
+    # ------------------------------------------------------------------------------------------
+
+    if command == 'stats':
+        if len(phonebook) == 1:
+            print(f'There is {len(phonebook)} contact in your phonebook')
+            print(phonebook)
+            command = ''
+            continue
+        else:
+            print(f'There are {len(phonebook)} contacts in your phonebook')
+            print(phonebook)
+            command = ''
+            continue
+
+    # ------------------------------------------------------------------------------------------
+
+    if command == 'add':
+        name = input('Type name: ')
+        number = input('Type number: ')
+        if name not in phonebook:
+            if number_validation(number):
+                phonebook[name] = number
+                with open('phonebook.json', 'w') as file:
```

Comment:
Change the modifier to `w+` to avoid an error if the file doesn't exist

Reply:
Changed. I will study the topic again

Reply:
Well, actually, it was my mistake. The `w` modifier works both for existing and non-existing files.